frontend/javascripts/oxalis/view/scalebar.js

```diff
@@ -22,34 +22,62 @@ type OwnProps = {|
 type StateProps = {|
   dataset: APIDataset,
   zoomValue: number,
-  widthInPixels: number,
-  heightInPixels: number,
+  viewportWidthInPixels: number,
+  viewportHeightInPixels: number,
 |};
 
 type Props = {|
   ...OwnProps,
   ...StateProps,
 |};
 
-const scalebarWidthPercentage = 0.25;
+const getBestScalebarAnchorInNm = (lengthInNm: number): number => {
+  const closestExponent = Math.floor(Math.log10(lengthInNm));
+  const closestPowerOfTen = 10 ** closestExponent;
+  const mantissa = lengthInNm / closestPowerOfTen;
 
-function Scalebar({ zoomValue, dataset, widthInPixels, heightInPixels }: Props) {
-  const widthInNm = convertPixelsToNm(widthInPixels, zoomValue, dataset);
-  const heightInNm = convertPixelsToNm(heightInPixels, zoomValue, dataset);
-  const formattedScalebarWidth = formatNumberToLength(widthInNm * scalebarWidthPercentage);
+  let bestAnchor = 1;
+  for (const anchor of [2, 5, 10]) {
+    if (Math.abs(anchor - mantissa) < Math.abs(bestAnchor - mantissa)) {
+      bestAnchor = anchor;
+    } else {
+      break;
+    }
+  }
+  return bestAnchor * closestPowerOfTen;
+};
+
+const scalebarWidthFactor = 0.3;
```

Comment:
```suggestion
// This factor describes how wide the scalebar would ideally be.
// However, this is only a rough guideline, as the actual width is changed
// so that round length values are represented.
const idealScalebarWidthFactor = 0.3;
```

Reply:
How much should I use comments in general? I'm trying to express as much as possible in code (via naming, etc.). Maybe we could talk about any conventions regarding this.